tests/integration/mixer/telemetry/metrics/blackhole/blackhole_test.go

```diff
@@ -0,0 +1,117 @@
+//  Copyright 2020 Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package blackhole
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment"
+	"istio.io/istio/pkg/test/framework/components/galley"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/mixer"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/components/sleep"
+	"istio.io/istio/pkg/test/framework/label"
+	"istio.io/istio/pkg/test/framework/resource"
+	util "istio.io/istio/tests/integration/mixer"
+)
+
+var (
+	sleepNs    namespace.Instance
+	ist        istio.Instance
+	prom       prometheus.Instance
+)
+
+func TestBlackHoleClusterMetric(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			sleepInst := sleep.DeployOrFail(t, ctx, sleep.Config{Namespace: sleepNs, Cfg: sleep.Sleep})
+			respCode, err := sleepInst.Curl("http://istio.io")
+			if err != nil {
+				t.Fatalf("Unable to exec curl http://istio.io from sleep pod: %v", err)
+			}
+			if respCode != "502" {
+				t.Fatalf("502 not returned from sleep pod; received http response code: %s", respCode)
+			}
+			query := `sum(istio_requests_total{destination_service_name="BlackHoleCluster"})`
+			util.ValidateMetric(t, prom, query, "istio_requests_total", 1)
+
+			respCode, err = sleepInst.Curl("https://istio.io")
```

Comment:
I suggest using https://github.com/golang/go/wiki/TableDrivenTests so that you can easily identify what failed

Reply:
There may be room for improvement with the way the modified tests are now in. I'll wait to resolve.